pkg/controllers/user/logging/utils/secret.go

```diff
@@ -0,0 +1,114 @@
+package utils
+
+import (
+	"reflect"
+
+	"github.com/pkg/errors"
+	rv1 "github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	loggingconfig "github.com/rancher/rancher/pkg/controllers/user/logging/config"
+)
+
+var (
+	initSecretData = map[string][]byte{
+		loggingconfig.CaFileName:     []byte(""),
+		loggingconfig.ClientCertName: []byte(""),
+		loggingconfig.ClientKeyName:  []byte(""),
+	}
+)
+
+func UpdateSSLAuthentication(prefix, target string, esConfig *v3.ElasticsearchConfig, spConfig *v3.SplunkConfig, kfConfig *v3.KafkaConfig, secrets rv1.SecretInterface) error {
+	var certificate, clientCert, clientKey string
```

Comment:
I feel like target is a bit redundant here. The config type can be determined by either of esConfig/spConfig/kfConfig not being nil